### DISCUSSION TITLE
Domain lookup fails on Android over wireguard VPN

### DISCUSSION THREAD

**Discussion by tomaThomas** [immich-app/immich#10802] [Q&A]
### The bug

The DNS lookup behaviour seems a bit odd.

When in the home network via wifi, dns lookup works fine using the routers DNS server.
However, when connected via wireguard VPN running on the router (with dns server set to router IP), immich is unable to resolve the domain, see the logs

Other apps are able to resolve the domain (Browser/Firefox, traceroute in Fing). Also, an entry in /etc/hosts is not used

### The OS that Immich Server is running on

Arch

### Version of Immich Server

v1.107.0

### Version of Immich Mobile App

v1.107.1

### Platform with the issue

- [ ] Server
- [ ] Web
- [X] Mobile

### Your docker-compose.yml content

```YAML
n/a
```

### Your .env content

```Shell
n/a
```

### Reproduction steps

```bash
1. Configure a domain in the router's dns server
2. Set up Wireguard VPN and connect android phone to it (from WAN side, using the Wireguard app)
3. Try to connect in the android app

On a rooted phone:
1. Put domain used in immich app in /etc/hosts with different IP
2. See immich app not connecting to the configured IP
```

### Relevant log output

```shell
ApiException 400: Socket operation failed: GET /albums (Inner exception: Failed host lookup: 'immich.home')

#0      IOClient.send (package:http/src/io_client.dart:94)
<asynchronous suspension>
#1      BaseClient._sendUnstreamed (package:http/src/base_client.dart:93)
<asynchronous suspension>
#2      ApiClient.invokeAPI (package:openapi/api_client.dart:101)
<asynchronous suspension>
#3      AlbumsApi.getAllAlbums (package:openapi/api/albums_api.dart:371)
<asynchronous suspension>
#4      AlbumService.refreshRemoteAlbums (package:immich_mobile/services/album.service.dart:154)
<asynchronous suspension>
#5      Future.wait.<anonymous closure> (dart:async/future.dart:524)
<asynchronous suspension>
```

**Comment by bo0tzz** [comment]
This seems like a network/dns issue, not an Immich one. I don't believe that using /etc/hosts is common at all on android, and I wouldn't be surprised if Flutter doesn't consider it at all. Wireguard should let you configure an actual DNS server to use.

**Reply by tomaThomas** [reply]
It's not common to use /etc/hosts, but it is respected by firefox and I guess the system dns resolver (AdAway uses it to block ad domains, which would be pointless if it's not used by default in all apps).
So if immich does not do anything special, this seems to be a flutter problem/behaviour

The DNS server configured in wireguard is not used by immich/flutter (but by firefox and other apps i tested)

Thanks for your reply!

**Reply by tomaThomas** [reply]
I've just set up a basic flutter app, and both of these worked over wireguard:
```
import 'dart:io';
import 'package:http/http.dart' as http;

InternetAddress.lookup("immich.home")
http.get(Uri.http('immich.home:3001', ''))
```
So it's not a general flutter problem
